Classes/ToxWrapper/OCTTox.m

```diff
@@ -697,6 +699,20 @@ - (BOOL)sendAvatarInfoToFriendNumber:(int32_t)friendNumber
     return (result == 0);
 }
 
+- (int)fileSendRequestWithFriendNumber:(int32_t)friendNumber fileName:(NSString *)fileName
+{
+    const char *cfileName = [fileName cStringUsingEncoding:NSUTF8StringEncoding];
+    uint16_t filename_length = [fileName lengthOfBytesUsingEncoding:NSUTF8StringEncoding];
+    
+    uint64_t fileSize = [[[NSFileManager defaultManager] attributesOfItemAtPath:fileName error:nil] fileSize];
```

Comment:
`fileName` isn't a path. OCTTox shouldn't know anything about file system. You can add `fileSize:(uint64_t)fileSize` parameter to this method.